### PR TITLE
VTCCA-2980 adding feature flag for new external-valuation-management-…

### PR DESCRIPTION
**Pull request by DanielConnelly**
…api call

### CODE REVIEW

File: app/config/GuiceModule.scala [hunk 1]
```diff
@@ -42,4 +48,15 @@ class GuiceModule(
     bind(classOf[Clock]).toInstance(Clock.systemUTC())
   }
 
+  protected def bindBoolean(path: String, name: String = ""): Unit =
+    bindConstant()
+      .annotatedWith(named(resolveAnnotationName(path, name)))
+      .to(Try(configuration.get[String](path).toBoolean).toOption.getOrElse(configException(path))) //We need to parse as string, due to the process of adding in from app-config-<env> it is seen as a string
+
+  private def resolveAnnotationName(path: String, name: String): String = name match {
```
Comment (darrell-plant) [inline app/config/GuiceModule.scala]: I don't know why we bother with `resolveAnnotationName`. Isn't it better to always provide a name?

**Comment by darrell-plant** [comment]
LGTM 👍

File: app/config/GuiceModule.scala [hunk 2]
```diff
@@ -42,4 +48,15 @@ class GuiceModule(
     bind(classOf[Clock]).toInstance(Clock.systemUTC())
   }
 
+  protected def bindBoolean(path: String, name: String = ""): Unit =
+    bindConstant()
+      .annotatedWith(named(resolveAnnotationName(path, name)))
+      .to(Try(configuration.get[String](path).toBoolean).toOption.getOrElse(configException(path))) //We need to parse as string, due to the process of adding in from app-config-<env> it is seen as a string
```
Comment (PeterPerhac) [inline app/config/GuiceModule.scala]: I think we need to do `configuration.get[Boolean](path)` and that will naturally explode with relevant ConfigurationException when the configuration isn't found under that path, or fails to be read (bad data type, etc.)

Comment (DanielConnelly) [reply app/config/GuiceModule.scala]: we need to get the string due to the nature of app-config-<env> passing the values.

Comment (PeterPerhac) [reply app/config/GuiceModule.scala]: configuration object implements `get` like so

```scala
  def get[A](path: String)(implicit loader: ConfigLoader[A]): A = {
    loader.load(underlying, path)
  }
```
so if you say get[Boolean] it defers the String->Boolean conversion to the ConfigLoader[Boolean]. I am not sure I follow, but surely we can configure boolean flags as non-strings?

app-config-... are yaml files, no? and YAML, for laughs, supports 12 different values for Boolean: https://yaml.org/type/bool.html

Comment (DanielConnelly) [reply app/config/GuiceModule.scala]: will discuss monday

**Comment by reddy-yattapu** [comment]
:+1:

**Pull request merged**